### PR TITLE
Added test with all non-primaries being isolated from each other.

### PR DESCRIPTION
**Pull request by HristoStaykov**
In this test we isolate the non-primary replicas form each other,
leaving only their connection to the primary. In this setup only
the primary is the only one who can bidirectionally communicate
with each replica. In this state the system can progress only up
to the working window size. We advance past the first stable checkpoint
which only the primary will collect and we initiate view change.
We perform this test in multiple cycles to verify correct recovery
and in the end we drop all adversaries and verify fast path is
recovered.

* **Problem Overview**  
 Additional test implemented for a corner case of the system's operation.
* **Testing Done**  
  NA

### CODE REVIEW

**Review by teoparvanov — APPROVED** [review]
Excellent test scenario, thanks Hristo!

**Review by titusen — APPROVED** [review]

File: tests/apollo/test_skvbc_restart_recovery.py [hunk 1]
```diff
@@ -662,6 +662,52 @@ async def test_recovering_view_after_restart_with_packet_loss(self, bft_network,
             await bft_network.wait_for_fast_path_to_be_prevalent(
             run_ops=lambda: skvbc.run_concurrent_ops(num_ops=20, write_weight=1), threshold=20)
 
+    @with_trio
+    @with_bft_network(start_replica_cmd, selected_configs=lambda n, f, c: f >= 2, rotate_keys=False)
+    async def test_isolated_non_primaries(self, bft_network):
```
[the remark this answers is not in the document]
Comment (HristoStaykov) [reply tests/apollo/test_skvbc_restart_recovery.py:668]: Good point, thanks for the reminder @titusen !
